src/main/java/net/dv8tion/jda/api/events/guild/scheduledevent/update/GuildScheduledEventUpdateLocationEvent.java

```diff
@@ -0,0 +1,273 @@
+     * {@link net.dv8tion.jda.api.entities.VoiceChannel VoiceChannel}, or at an external location not necessarily in
+     * a guild. This wrapper contains methods for identifying the {@link GuildScheduledEvent.Type Type} of location as well retrieving the location itself.
+     * Note: In some cases the {@link GuildScheduledEvent.Type Type} of the location may be {@link GuildScheduledEvent.Type#UNKNOWN}, which indicates that
+     * Discord has added a new guild scheduled event type, and the current JDA version being used does not yet support it.
+     *
+     * <h2> Example </h2>
+     * <pre>{@code
+     * Location location = guildScheduledEventUpdateLocationEvent.getNewLocation();
+     * GuildScheduledEvent.Type locationType = location.getType();
+     * String logMessage = "";
+     * switch (locationType)
+     * {
+     *     case EXTERNAL:
+     *         String externalLocation = location.getExternalLocation();
+     *         logMessage = "This event will now take place at " + externalLocation;
+     *         break;
+     *     case STAGE_INSTANCE:
+     *         StageChannel stageChannel = location.getStageChannel();
+     *         logMessage = "This event will now take place at the following Stage Channel: " + stageChannel.getAsMention();
+     *         break;
+     *     case VOICE:
+     *         VoiceChannel voiceChannel = location.getVoiceChannel();
+     *         logMessage = "This event will now take place at the following Voice Channel: " + voiceChannel.getAsMention();
+     *         break;
+     *     case UNKNOWN:
+     *         logMessage = "Sorry, this bot cannot determine the location of the event";
+     * }
+     * }</pre>
+     */
+    public static class Location
+    {
+        // Only one of these will not be null at any given time. If all are null, then the event's location is unknown to JDA.
+        private String externalLocation;
+        private StageChannel stageChannel;
+        private VoiceChannel voiceChannel;
+
+        /**
+         * Constructs a new {@link Location} object from the provided object.
+         *
+         * @param object The location of the event.
+         *               <br> The object must be either an instance of {@link StageChannel},
+         *               {@link VoiceChannel}, or {@link String} in the case of an external locations. A {@code null}
+         *               object indicates that the event's location {@link GuildScheduledEvent.Type Type} is
+         *               {@link GuildScheduledEvent.Type#UNKNOWN Type#UNKNOWN}.
+         */
+        public Location(@Nullable Object object)
+        {
+            if (object instanceof String)
+                externalLocation = (String) object;
+            else if (object instanceof StageChannel)
+                stageChannel = (StageChannel) object;
+            else if (object instanceof VoiceChannel)
+                voiceChannel = (VoiceChannel) object;
+        }
```

Comment:
I'd personally use factory methods instead, ofString, ofStage, ofVoice etc

This looks really, weird, and also accepts random Object types.

Reply:
That was my intention, but looking back on that I agree with Tais it does seem very strange and I'm not really sure why I written it that way. I feel like public constructors or factory methods may wrongly give the impression that the Location wrapper object would need to be created by the user, perhaps we could turn this into an inner interface and implement it internally?

Reply:
Update: Upon further inspecting the code it seems like this constructor is redundant anyways, and can just be deleted.

Reply:
yea you are right, we don't actrually use this one at all. We do use "public Location(@Nonnull GuildScheduledEvent guildScheduledEvent)"